doc/sonic-installer/SONiC_Image_Installation_Verification.md

```diff
@@ -0,0 +1,282 @@
+
+The verification should looks like this:
+```
+admin@vlab-01:~$ sudo sonic-installer install sonic-broadcom.bin -y
+Image file 'sonic-broadcom.bin' is of a different platform ASIC type than running platform's.
+If you are sure you want to install this image, use --skip-platform-check.
+Aborting...
+Aborted!
+```
+
+# 2 Functional Requirements
+
+- sonic-installer should be able to verify if an image is valid to be installed on runnning platform.
+- sonic-installer should provide a way to skip such platform ASIC type check and an option of force image installation
+- sonic-intaller should not affect the past images installation. 
+
+# 3 Configuration and Management Requirements
+
+- The option of platform AISC type check skip should be added to sonic-installer help menu.
+  - Use `--skip-platform-check` option instead of `--force` as the later one has been used for image force installation which skips image secure boot check.
+```
+admin@vlab-01:~$ sudo sonic-installer install --help
+Usage: sonic-installer install [OPTIONS] URL
+
+  Install image from local binary or URL
+
+Options:
+  -y, --yes
+  -f, --force, --skip-secure-check
+                                  Force installation of an image of a non-
+                                  secure type than secure running image
+  --skip-platform-check           Force installation of an image of of a different       <======
+                                  ASIC type from the current platform
+  --skip_migration                Do not migrate current configuration to the
+                                  newly installed image
+  --skip-package-migration        Do not migrate current packages to the newly
+                                  installed image
+  --skip-setup-swap               Skip setup temporary SWAP memory used for
+                                  installation
+  --swap-mem-size INTEGER         SWAP memory space size  [default: (1024
+                                  MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --total-mem-threshold INTEGER   If system total memory is lower than
+                                  threshold, setup SWAP memory  [default:
+                                  (2048 MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --available-mem-threshold INTEGER
+                                  If system available memory is lower than
+                                  threhold, setup SWAP memory  [default: (1200
+                                  MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --help                          Show this message and exit.
+
+```
+
+# 4 Design and Flow Diagrams
+
+## 4.1 Flow Diagrams
+The following figure shows how to intall an image through SONiC CLI.
```

Comment:
>figure [](http://example.com/codeflow?start=14&length=6)

In the figure, the feedback loop to CLI does not make sense, it should connect to anther terminator.

I feel the 'Image Verification' is too broad. Break it down? At least you need to get image ASIC and runtime ASIC info. Is the flow same in SONiC env and in ONIE env?

Reply:
Yes. The overall flow is the same. Getting current platform then grep to see if it exists in runtime ASIC platforms list.
The different part is that in ONIE, the comparison happens in `installer` folder(`if ! grep -Fxq "$onie_platform" platforms_asic;`).
In sonic, the comparison happens as we are facing bin file, so we need to extract platforms_asic.(`["tar", "xf", "-", PLATFORMS_ASIC, "-O"]`, then `["grep", "-Fxq", "-m 1", platform]`).

Reply:
What do you mean `connect to another terminator`. Let's say we are in SONiC CLI, if the verification failed, we just back to our CLI, right?